File: app/src/main/java/com/varunbarad/popularmovies/model/data/VideoList.java
package com.varunbarad.popularmovies.model.data;

import com.google.gson.Gson;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

/**
 * Creator: vbarad
 * Date: 2016-12-07
 * Project: PopularMovies
 */

public final class VideoList {
  @Expose
  @SerializedName("results")
  private ArrayList<Video> results;

  /**
   * Public no-args constructor for serialization
   */
  public VideoList() {

  }

  /**
   * @param results
   */
  public VideoList(final ArrayList<Video> results) {
    this.results = results;
  }
  
  public ArrayList<Video> getResults() {
    return results;
  }

  @Override
  public String toString() {
    return (new Gson()).toJson(this);
  }
}

File: app/src/main/java/com/varunbarad/popularmovies/model/data/VideoList.kt
package com.varunbarad.popularmovies.model.data

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import com.squareup.moshi.Moshi

@JsonClass(generateAdapter = true)
data class VideoList(
    @Json(name = "results") val results: List<Video>
) {
    override fun toString(): String {
        return Moshi.Builder().build().adapter(VideoList::class.java).toJson(this)
    }
}
